skimage/color/tests/test_colorconv.py

```diff
@@ -183,32 +184,33 @@ def test_xyz_rgb_roundtrip(self):
         img_rgb = img_as_float(self.img_rgb)
         assert_array_almost_equal(xyz2rgb(rgb2xyz(img_rgb)), img_rgb)
 
-    # RGB<->HED roundtrip with ubyte image
+    # HED<->RGB roundtrip with ubyte image
     def test_hed_rgb_roundtrip(self):
-        img_rgb = img_as_ubyte(self.img_rgb)
-        new = img_as_ubyte(hed2rgb(rgb2hed(img_rgb)))
-        assert_equal(new, img_rgb)
+        img_in = img_as_ubyte(self.img_stains)
+        img_out = rgb2hed(hed2rgb(img_in))
+        assert_equal(img_as_ubyte(img_out), img_in)
 
-    # RGB<->HED roundtrip with float image
+    # HED<->RGB roundtrip with float image
     def test_hed_rgb_float_roundtrip(self):
-        img_rgb = img_as_float(self.img_rgb)
-        assert_array_almost_equal(hed2rgb(rgb2hed(img_rgb)), img_rgb)
-
-    # RGB<->HDX roundtrip with ubyte image
-    def test_hdx_rgb_roundtrip(self):
-        from skimage.color.colorconv import hdx_from_rgb, rgb_from_hdx
-        img_rgb = self.img_rgb
-        conv = combine_stains(separate_stains(img_rgb, hdx_from_rgb),
-                              rgb_from_hdx)
-        assert_equal(img_as_ubyte(conv), img_rgb)
-
-    # RGB<->HDX roundtrip with float image
-    def test_hdx_rgb_roundtrip_float(self):
-        from skimage.color.colorconv import hdx_from_rgb, rgb_from_hdx
-        img_rgb = img_as_float(self.img_rgb)
-        conv = combine_stains(separate_stains(img_rgb, hdx_from_rgb),
-                              rgb_from_hdx)
-        assert_array_almost_equal(conv, img_rgb)
+        img_in = self.img_stains
+        img_out = rgb2hed(hed2rgb(img_in))
+        assert_array_almost_equal(img_out, img_in)
+
+    # BRO<->RGB roundtrip with ubyte image
+    def test_bro_rgb_roundtrip(self):
+        from skimage.color.colorconv import bro_from_rgb, rgb_from_bro
+        img_in = img_as_ubyte(self.img_stains)
+        img_out = combine_stains(img_in, rgb_from_bro)
+        img_out = separate_stains(img_out, bro_from_rgb)
+        assert_equal(img_as_ubyte(img_out), img_in)
+
+    # BRO<->RGB roundtrip with float image
+    def test_bro_rgb_roundtrip_float(self):
+        from skimage.color.colorconv import bro_from_rgb, rgb_from_bro
+        img_in = self.img_stains
+        img_out = combine_stains(img_in, rgb_from_bro)
+        img_out = separate_stains(img_out, bro_from_rgb)
+        assert_array_almost_equal(img_out, img_in)
```

Comment:
The new RGB/BRO tests look good and it doesn't seem these functions were previously being tested. 

Ideally we would maintain some kind of test of `rgb_from_hdx` and `hdx_from_rgb`, as those don't seem to be tested elsewhere. Maybe just keep a round trip test, but without the combine/separate stains calls?

Reply:
`rgb_from_hdx` and `hdx_from_rgb` are constants, and so are the two arrays I replaced them with: https://github.com/scikit-image/scikit-image/blob/ff188272a0ea509cb4d6025d1906c494914a4304/skimage/color/colorconv.py#L485

There's a long list of these arrays there. These arrays describe the RGB color of each of the stains in a common set. If you want to test that these arrays are correct, you'd have to source an image of a slide stained with the corresponding stains, apply the stain unmixing, and verify that there are few negative values in the result.

It is the functions [`separate_stains`](https://github.com/scikit-image/scikit-image/blob/ff188272a0ea509cb4d6025d1906c494914a4304/skimage/color/colorconv.py#L1391) and [`combine_stains`](https://github.com/scikit-image/scikit-image/blob/ff188272a0ea509cb4d6025d1906c494914a4304/skimage/color/colorconv.py#L1459) that we're testing here with these round-trips. Both the HED<->RGB and the BRO<->RGB round-trips test the same two functions, just with different data.

If you really want to test these functions with each of the data matrices, we could certainly do that. For the 2-stain varieties, we'd have to compare only the first two channels after the roundtrip. Or set the 3rd channel of the input to 0. I just don't think that such a test would add anything here.

Reply:
I mean we could just verify that `rgb_from_hdx @ hdx_from_rgb` is the identity matrix to ensure that the pair of matrices would correctly reverse each other. But this is guaranteed because the one is computed as the inverse of the other.

Reply:
Sorry for the noise on this comment. I was indeed thinking these were functions and not just the transform matrices. This seems fine as-is, then.

Reply:
No need to apologize. I appreciate your thoroughness. Interactions I've had with this project have all given me a really good impression. Keep up the good work!